Src/FluentAssertions/Primitives/TimeOnlyAssertions.cs

```diff
@@ -127,6 +128,112 @@ public AndConstraint<TAssertions> Be(TimeOnly? expected, string because = "", pa
         return new AndConstraint<TAssertions>((TAssertions)this);
     }
 
+    /// <summary>
+    /// Asserts that the current <see cref="TimeOnly"/>  is within the specified time
+    /// from the specified <paramref name="nearbyTime"/> value.
+    /// </summary>
+    /// <param name="nearbyTime">
+    /// The expected time to compare the actual value with.
+    /// </param>
+    /// <param name="precision">
+    /// The maximum amount of time which the two values may differ.
+    /// </param>
+    /// <param name="because">
+    /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+    /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+    /// </param>
+    /// <param name="becauseArgs">
+    /// Zero or more objects to format using the placeholders in <paramref name="because" />.
+    /// </param>
+    public AndConstraint<TAssertions> BeCloseTo(TimeOnly nearbyTime, TimeSpan precision, string because = "",
+        params object[] becauseArgs)
+    {
+        if (precision < TimeSpan.Zero)
+        {
+            throw new ArgumentOutOfRangeException(nameof(precision), $"The value of {nameof(precision)} must be non-negative.");
+        }
+
+        if (precision == TimeSpan.MaxValue)
+        {
+            precision = precision.AddTicks(-1);
+        }
+
+        TimeOnly minimumValue = nearbyTime.Add(-precision);
+        TimeOnly maximumValue = nearbyTime.Add(precision.AddTicks(1));
+
+        TimeSpan? difference = (Subject != null)
+            ? MinimumDifference(Subject.Value, nearbyTime)
+            : null;
+
+        Execute.Assertion
+            .BecauseOf(because, becauseArgs)
+            .WithExpectation("Expected {context:the time} to be within {0} from {1}{reason}, ", precision, nearbyTime)
+            .ForCondition(Subject is not null)
+            .FailWith("but found <null>.")
+            .Then
+            .ForCondition(Subject.Value.IsBetween(minimumValue, maximumValue))
```

Comment:
If a call to `BeCloseTo` is wrapped inside an `AssertionScope` this chain of calls continues even `.ForCondition(Subject is not null)` fails.
```suggestion
            .ForCondition(Subject?.IsBetween(minimumValue, maximumValue) == true)
```

Reply:
Ok.. don't get why? But cool to know :D

Reply:
Just re-thought this.. and yes.. now I get it. This is just the "normal" use-case for an assertion scope